tests/installation/bootloader_hyperv.pm

```diff
@@ -36,44 +36,35 @@ sub run {
-
-    # Copy assets from NFS to Hyper-V cache
-    for my $n ('', 1 .. 9) {
-        # Look for {ISO,HDD}, {ISO,HDD}_1, ... variables
-        $n = "_$n" if $n;
-        if (my $iso = get_var("ISO$n")) {
-            for my $isopath ("iso", "iso\\fixed") {
-                # Copy ISO from NFS share to local cache on Hyper-V in 'network-restartable' mode
-                my $basenameiso = basename($iso);
-                # Using sha256sum to judge whether file is used by other job process.
-                hyperv_cmd_with_retry("if exist $root\\cache\\$basenameiso ( sha256sum $root\\cache\\$basenameiso )", {msgs => ('Permission denied')});
-                last
-                  unless hyperv_cmd("if not exist $root\\cache\\$basenameiso ( copy /Z /Y $root_nfs\\$isopath\\$basenameiso $root\\cache\\ )",
-                    {ignore_return_code => 1});
+    my $mount_result = hyperv_cmd_with_retry("if not exist N: ( mount \\\\openqa.suse.de\\var\\lib\\openqa\\share\\factory N: )",
+        {msgs => ('Access is denied.')});
+
+    # Main script logic
+    if ($mount_result->{success}) {
+        for my $n ('', 1 .. 9) {
+            $n = "_$n" if $n;
+            if (my $iso = get_var("ISO$n")) {
+                copy_iso_to_cache($iso, $root, $root_nfs, $svirt);
+            }
+            if (my $hdd = get_var("HDD$n")) {
+                copy_hdd_to_cache($hdd, $root, $root_nfs, $svirt);
             }
         }
-        if (my $hdd = get_var("HDD$n")) {
-            my $basenamehdd = basename($hdd);
-            for my $hddpath ("hdd", "hdd\\fixed") {
-                my $basenamehdd_vhd = $basenamehdd =~ s/vhdx\.xz/vhdx/r;
-                # If the image exists, do nothing
-                last if hyperv_cmd("if exist $root\\cache\\$basenamehdd_vhd ( exit 1 )", {ignore_return_code => 1});
-                # Copy HDD from NFS share to local cache on Hyper-V
-                hyperv_cmd_with_retry("copy /z $root_nfs\\$hddpath\\$basenamehdd $root\\cache\\");
-                # Decompress the XZ compressed image
-                if ($hdd =~ m/vhdx\.xz/) {
-                    record_info 'unxz', "Decompressing $root\\cache\\$basenamehdd";
-                    my ($ret, $stdout, $stderr) = $svirt->run_cmd("xz --decompress --keep --verbose $root\\cache\\$basenamehdd", wantarray => 1);
-                    defined($stderr) && $stderr =~ /xz: $root\\cache\\$basenamehdd: File exists/ && sleep 60;
-                    last;
-                }
+    } else {
+        my $iso_base_url = "http://openqa.suse.de/assets/iso/";
+        my $hdd_base_url = "http://openqa.suse.de/assets/hdd/";
+
+        for my $n ('', 1 .. 9) {
+            $n = "_$n" if $n;
+            if (my $iso = get_var("ISO$n")) {
+                download_iso_from_http($iso, $root, $iso_base_url, $svirt);
+            }
+            if (my $hdd = get_var("HDD$n")) {
+                download_hdd_from_http($hdd, $root, $hdd_base_url, $svirt);
```

Comment:
No return value from download_hdd_from_http, download_iso_from_http, copy_iso_to_cache and copy_iso_to_cache ?